sql/hive/src/test/scala/org/apache/spark/sql/sources/hadoopFsRelationSuites.scala

```diff
@@ -668,6 +671,42 @@ abstract class HadoopFsRelationTest extends QueryTest with SQLTestUtils with Tes
       df.write.format(dataSourceName).partitionBy("c", "d", "e").saveAsTable("t")
     }
   }
+
+  test("Locality support for FileScanRDD") {
```

Comment:
How about we add the following test cases?

* Every partition of the FileScanRDD has a single file.
* There are some large files. So, there are some partitions and every such a partition has a part of a file.
* If we are going to pick the top K locations that host the largest amount of data, let's also add tests.